packages/eas-cli/src/commands/branch/list.ts

```diff
@@ -18,6 +18,59 @@ import { findProjectRootAsync, getProjectFullNameAsync } from '../../project/pro
 import { getActorDisplayName } from '../../user/actions';
 
 const BRANCHES_LIMIT = 10_000;
+export async function listBranchesAsync({
+  fullName,
+}: {
+  fullName: string;
+}): Promise<
+  (Pick<UpdateBranch, 'id' | 'name'> & {
+    updates: (Pick<Update, 'id' | 'updatedAt' | 'message'> & {
+      actor?: Maybe<Pick<User, 'username' | 'id'> | Pick<Robot, 'firstName' | 'id'>>;
+    })[];
+  })[]
+> {
+  const data = await withErrorHandlingAsync(
+    graphqlClient
+      .query<BranchesByAppQuery, BranchesByAppQueryVariables>(
+        gql`
+          query BranchesByAppQuery($fullName: String!, $limit: Int!) {
+            app {
+              byFullName(fullName: $fullName) {
+                id
+                fullName
+                updateBranches(offset: 0, limit: $limit) {
+                  id
+                  name
+                  updates(offset: 0, limit: 1) {
+                    id
+                    actor {
+                      __typename
+                      id
+                      ... on User {
+                        username
+                      }
+                      ... on Robot {
+                        firstName
+                      }
+                    }
+                    updatedAt
+                    message
+                  }
+                }
+              }
+            }
+          }
+        `,
+        {
+          fullName,
+          limit: BRANCHES_LIMIT,
+        }
+      )
+      .toPromise()
+  );
+
+  return data?.app?.byFullName.updateBranches ?? [];
+}
```

Comment:
this is unchanged, just pulled out of the class below so that it can be exported.